articles/best-practices/rules.md

```diff
@@ -0,0 +1,54 @@
+---
+description: Best practices for writing and managing Auth0 Rules.
+topics:
+  - best-practices
+  - rules
+contentType: 
+    - index
+    - reference
+useCase:
+  - best-practices
+  - rules
+---
+
+# Best Practices: Rules
+
+In this article, you'll learn some best practices for using [rules](/rules). Before you start writing rules, review [what you can use rules for](/current#what-can-i-use-rules-for-) and take a look at some [examples](/rules/current#examples)
+
+## Handle errors
+
+Make sure your rules code catches errors after calls which may trigger an error. Also, ensure every branch through the code ends with a return statement to call the callback.
+
+## Review rule order
+
+Rules execute in the order shown on the Auth0 Dashboard. If a rule depends on the execution of another rule, move the dependent rule lower in the rules list.
+
+## Exit soon
+
+To improve performance, write rules that exit as soon as possible.
+
+For example, if a rule has 3 checks to decide if it should run, the first check should eliminate the most cases. Followed by the check that eliminates the second-highest number of cases for the rule to run, and so on.
+
+## Reduce API requests
+
+Try not to use a lot of API calls in rules. Too many can slow down login response time and may cause failures during a timeout.
+
+Avoid calling the Management API in if possible, especially in high volume environments.
+
+## Cache results
+
+Rules have a [global variable you can use to cache information](/rules/current#cache-expensive-resources). Use it when making non-user specific API calls and cache the results between users. For example, getting an access token to your API.
```

Comment:
Unimportant grammar observation/question - is it `non-user specific` or `non user specific` or `non-user-specific` or `non user-specific`? I usually see/write a hyphenated `user-specific`, but I couldn't make a combination work right unless the entire phrase was rearranged. Thoughts?